src/LibraryManager.Vsix/Shared/LibraryHelpers.cs

```diff
@@ -128,12 +128,20 @@ public static bool IsSupported(this Project project)
             Manifest manifest = await Manifest.FromFileAsync(configFileName, dependencies, CancellationToken.None).ConfigureAwait(false);
             var hostInteraction = dependencies.GetHostInteractions() as HostInteraction;
 
-            int? filesDeleted = manifest?.Clean((file) => hostInteraction.DeleteFiles(file));
+            IEnumerable<ILibraryInstallationResult> results = manifest?.Clean((file) => hostInteraction.DeleteFiles(file));
 
-            Logger.LogEvent(Resources.Text.CleanLibrariesSucceeded + Environment.NewLine, LogLevel.Task);
+            if (results != null && results.All(r => r.Success))
+            {
+                Logger.LogEvent(Resources.Text.CleanLibrariesSucceeded + Environment.NewLine, LogLevel.Task);
+            }
+            else
+            {
+                Logger.LogEvent(Resources.Text.CleanLibrariesFailed + Environment.NewLine, LogLevel.Task);
+            }
 
-            TelemetryResult result = configProjectItem != null ? TelemetryResult.Success : TelemetryResult.Failure;
-            Telemetry.TrackUserTask("clean", new KeyValuePair<string, object>("filesdeleted", filesDeleted));
+            // Do we need this? We could log the individual errors for each library instead
```

Comment:
Design question: Please comment!

Reply:
We should keep the telemetry here.  We also need to make sure that the `TelemetryResult` is getting hooked up correctly.  It looks like `TrackUserTask` needs an overload to accommodate the result.

Reply:
Can we also include error codes encountered in the failing cases?